File: SPOJ/MIRRORED.py
def isMirrored(a, b):
	clause_1 = (a == 'p' and b == 'q')
	clause_2 = (a == 'q' and b == 'p')
	clause_3 = (a == 'd' and b == 'b')
	clause_4 = (a == 'b' and b == 'd')
	if clause_1 or clause_2 or clause_3 or clause_4:
		return True
	else:
		return False

def main():
	print "Ready"
	while True:
		string = raw_input()
		a, b = string[0], string[1]
		if a == ' ' and b == ' ':
			break
		flag = isMirrored(a, b)
		if flag:
			print "Mirrored pair"
		else:
			print "Ordinary pair"

main()

File: SPOJ/MIRRORED.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>
#include <ctype.h>

bool process(char first, char second) {
	if(first == 'p' && second == 'q')
		return true;
	else if(first == 'q' && second == 'p')
		return true;
	else if(first == 'b' && second == 'd')
		return true;
	else if(first == 'd' && second == 'b')
		return true;
	else
		return false;
}

int main(int argc, char const *argv[]) {
	size_t size = 2;
	char* input;
	printf("Ready\n");
	while(true) {
		getline(&input, &size, stdin);
		char first = input[0], second = input[1];
		if(first == ' ' && second == ' ')
			break;
		bool ans = process(first, second);
		if(ans) {
			printf("Mirrored pair\n");
		}
		else if(!ans) {
			printf("Ordinary pair\n");
		}
	} 
	return 0;
}